src/ilasm/main.cpp

```diff
@@ -66,7 +69,7 @@ void MakeProperSourceFileName(__in __nullterminated WCHAR* wzOrigName,
         if(wzProperName[j] == '.') break;
         if((wzProperName[j] == '\\')||(j == 0))
         {
-            wcscat_s(wzProperName,MAX_FILENAME_LENGTH,L".il");
+            wcscat_s(wzProperName,MAX_FILENAME_LENGTH,W(".il"));
```

Comment:
Could you please create a github issue to eventually fix the filename size limits in ilasm? We have done similar thing in PAL / CoreCLR, since Unix supports much longer filenames.